modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/operation/ReflectiveMethodOperationExecutor.java

```diff
@@ -47,21 +48,31 @@
 
   private static final Logger LOGGER = getLogger(ReflectiveMethodOperationExecutor.class);
 
-  private final ReflectiveMethodComponentExecutor<M> executor;
+  final ReflectiveMethodComponentExecutor<M> executor;
   private MuleContext muleContext;
+  private BiFunction<Object, ExecutionContext<M>, Object> resultTransformer = (o, e) -> o;
 
   public ReflectiveMethodOperationExecutor(M operationModel, Method operationMethod, Object operationInstance) {
-    executor =
+    this.executor =
         new ReflectiveMethodComponentExecutor<>(operationModel.getParameterGroupModels(), operationMethod, operationInstance);
   }
 
+  public ReflectiveMethodOperationExecutor(M operationModel, Method operationMethod, Object operationInstance,
+                                           BiFunction<Object, ExecutionContext<M>, Object> resultTransformer) {
+    this.executor =
+        new ReflectiveMethodComponentExecutor<>(operationModel.getParameterGroupModels(), operationMethod, operationInstance);
+    this.resultTransformer = resultTransformer;
+  }
+
   /**
    * {@inheritDoc}
    */
   @Override
   public Publisher<Object> execute(ExecutionContext<M> executionContext) {
     try {
-      return justOrEmpty(executor.execute(executionContext));
+      Object result = executor.execute(executionContext);
+      result = resultTransformer.apply(result, executionContext);
```

Comment:
you just moved the transformer from the mediator here.

Reply:
yes, that is on purpose to be executed inside the umbrella of any policy applied to the method execution, this will let to the operation be executed again in case of a connectivity error.